src/main/scala/com/fulcrumgenomics/vcf/AssessPhasing.scala

```diff
@@ -0,0 +1,780 @@
+  def getOverlaps(locatable: Locatable): Set[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaSet
+    this.detectors(locatable.getContig).getOverlaps(locatable).toSet
+  }
+
+  def getAll: Iterator[PhaseBlock] = {
+    import scala.collection.JavaConversions.asScalaIterator
+    this.detectors.values.flatMap(_.getAll.iterator).toIterator
+  }
+}
+
+object PhaseBlock extends LazyLogging {
+  import scala.collection.mutable
+
+  /** Creates an overlap detector for blocks of phased variants.  Variants from the same block are found using the
+    * "PS" tag.
+    */
+  private[vcf] def toOverlapDetector(iterator: Iterator[VariantContext], dict: SAMSequenceDictionary): OverlapDetectorByContig = {
+    val detector = new OverlapDetectorByContig(dict=dict)
+    val progress = new ProgressLogger(logger)
+
+    logger.info("Creating the overlap detector")
+    // create the blocks
+    var curContig = ""
+    val phaseBlocks = mutable.HashMap[Int, PhaseBlock]()
+    while (iterator.hasNext) {
+      val ctx = iterator.next()
+      val phaseSetId: Int = PhasingUtils.getPhasingSetId(ctx)
+      if (phaseSetId > 0) {
+        if (ctx.getContig != curContig) {
+          detector.addAll(phaseBlocks.values.toIterator)
+          phaseBlocks.clear()
+          curContig = ctx.getContig
+        }
+        val phaseBlock = phaseBlocks.get(phaseSetId) match {
+          case Some(block) =>
+            require(block.getContig == ctx.getContig)
+            require(block.getStart <= ctx.getStart)
+            new PhaseBlock(contig=ctx.getContig, start=block.getStart, end=Math.max(block.getEnd, ctx.getEnd)) // extend the block
+          case None        =>
+            new PhaseBlock(contig=ctx.getContig, start=ctx.getStart, end=ctx.getEnd)
+        }
+        phaseBlocks.put(phaseSetId, phaseBlock)
+      }
+      progress.record(ctx.getContig, ctx.getStart)
+    }
+
+    // add any remaining
+    detector.addAll(phaseBlocks.values.toIterator)
+    phaseBlocks.clear()
+
+    detector
+  }
+}
+
+case class PhaseBlock private (contig: String, start: Int, end: Int) extends Locatable {
+  def getContig: String = contig
+  def getStart: Int = start
+  def getEnd: Int = end
+  def length: Int = end - start + 1
+}
+
+object AssessPhasingMetric {
+  val MetricExtension = ".assess_phasing_metrics.txt"
+}
+
+/** Some counts about phasing
+  *
+  * @param num_called the number of variants called.
+  * @param num_phased the number of variants called with phase.
+  * @param num_truth the number of variants with known truth genotypes.
+  * @param num_truth_phased the number of variants with known truth genotypes with phase.
+  * @param num_called_with_truth_phased the number of variants called that had a known phased genotype.
+  * @param num_phased_with_truth_phased the number of variants called with phase that had a known phased genotype.
+  * @param num_truth_phased_in_called_block the number of known phased variants that were in a called phased block.
```

Comment:
It implies that this known variant overlaps the span of a phase block.  The variant does not need to be called.